### PR TITLE
Do not strip leading zeros from trace IDs and span IDs

### PR DESCRIPTION
**Pull request by tobiasstadler**
## Which problem is this PR solving?
- Implements jaegertracing/jaeger#1657

### CODE REVIEW

File: src/jaegertracing/TraceID.h [hunk 1]
```diff
@@ -43,13 +43,8 @@ class TraceID {
...
     void print(Stream& out) const
     {
-        if (_high == 0) {
-            out << std::hex << _low;
-        }
-        else {
-            out << std::hex << _high << std::setw(16) << std::setfill('0')
-                << std::hex << _low;
-        }
+        out << std::setw(16) << std::setfill('0') << std::hex << _high
```
Comment (yurishkuro) [inline src/jaegertracing/TraceID.h:46]: this does not seem backwards compatible, what's the reason for removing `if (_high == 0) {`?

**Review by yurishkuro — COMMENTED** [review]
Q: since you're looking into this, I just want too make sure there is a unit test that ensures that this library still CAN consume non-padded IDs from the incoming request headers

**Comment by tobiasstadler** [comment]
> Q: since you're looking into this, I just want too make sure there is a unit test that ensures that this library still CAN consume non-padded IDs from the incoming request headers

TEST(SpanContext, testFromStream) tests both padded and unpadded trace ids

File: src/jaegertracing/TraceID.h (hunk 1 above)
Comment (tobiasstadler) [reply src/jaegertracing/TraceID.h]: done

Comment (tobiasstadler) [reply src/jaegertracing/TraceID.h]: I though the trace id is now always 32 hex chars long. But I was wrong.

Comment (yurishkuro) [reply src/jaegertracing/TraceID.h]: No, all Jaeger SDKs still use 64bit by default, you have to explicitly enable 128bit.

**Review by yurishkuro — APPROVED** [review]
Thanks!

File: src/jaegertracing/TraceID.h (hunk 1 above)
Comment (tobiasstadler) [reply src/jaegertracing/TraceID.h]: Yes, noticed that after your comment. I misread the prs for the other sdks.

**Pull request merged**